src/main.rs

```diff
@@ -256,207 +264,183 @@ const APP: () = {
-                phase: hal::spi::Phase::CaptureOnSecondTransition,
-            })
-            .manage_cs()
-            .suspend_when_inactive()
-            .cs_delay(220e-9);
-
-            dma_channels.0.set_peripheral_address(
-                &dp.SPI2.txdr as *const _ as u32,
-                false,
-            );
-            dma_channels
-                .0
-                .set_memory_address(&SPI_START as *const _ as u32, false);
-            dma_channels
-                .0
-                .set_direction(hal::dma::Direction::MemoryToPeripherial);
-            dma_channels.0.set_transfer_length(1);
-            dma_channels.0.cr().modify(|_, w| {
-                w.circ()
-                    .enabled()
-                    .psize()
-                    .bits16()
-                    .msize()
-                    .bits16()
-                    .pfctrl()
-                    .dma()
-            });
-            dma_channels.0.dmamux().modify(|_, w| {
-                w.dmareq_id()
-                    .variant(hal::stm32::dmamux1::ccr::DMAREQ_ID_A::TIM2_UP)
-            });
-
-            let mut spi: hal::spi::Spi<_, _, u16> = dp.SPI2.spi(
-                (spi_sck, spi_miso, hal::spi::NoMosi),
-                config,
-                50.mhz(),
-                ccdr.peripheral.SPI2,
-                &ccdr.clocks,
-            );
+        let mut sampling_timer = sampling_timer::SamplingTimer::new(timer2);
+        let sampling_timer_channels = sampling_timer.channels();
 
-            // Kick-start the SPI transaction - we will add data to the TXFIFO to read from the ADC.
-            let spi_regs = unsafe { &*hal::stm32::SPI2::ptr() };
-            spi_regs.cr1.modify(|_, w| w.cstart().started());
+        // Configure the SPI interfaces to the ADCs and DACs.
+        let adcs = {
+            let adc0 = {
+                let spi_miso = gpiob
+                    .pb14
+                    .into_alternate_af5()
+                    .set_speed(hal::gpio::Speed::VeryHigh);
+                let spi_sck = gpiob
+                    .pb10
+                    .into_alternate_af5()
+                    .set_speed(hal::gpio::Speed::VeryHigh);
+                let _spi_nss = gpiob
+                    .pb9
+                    .into_alternate_af5()
+                    .set_speed(hal::gpio::Speed::VeryHigh);
 
-            spi.listen(hal::spi::Event::Rxp);
+                let config = hal::spi::Config::new(hal::spi::Mode {
+                    polarity: hal::spi::Polarity::IdleHigh,
+                    phase: hal::spi::Phase::CaptureOnSecondTransition,
+                })
+                .manage_cs()
+                .suspend_when_inactive()
+                .cs_delay(220e-9);
 
-            spi
-        };
+                let spi: hal::spi::Spi<_, _, u16> = dp.SPI2.spi(
+                    (spi_sck, spi_miso, hal::spi::NoMosi),
+                    config,
+                    50.mhz(),
```

Comment:
Single definition for this too.

Reply:
Any reason why you're advocating for moving it out to a singular definition?

For context, the 50MHz is the maximum clock frequency allowed by the DACs.

Reply:
Moved to a new `design_parameters.rs` file containing other constants. Will likely move things out there too to clean up `main.rs` in the future.